File: bindings/csharp/src/ProvenVoip.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// VoIP/SIP protocol types for proven-servers.

namespace Proven;

/// <summary>Method matching the Idris2 ABI tags (0-12).</summary>
public enum Method : byte
{
    Invite = 0,
    Ack = 1,
    Bye = 2,
    Cancel = 3,
    Register = 4,
    Options = 5,
    Info = 6,
    Update = 7,
    Subscribe = 8,
    Notify = 9,
    Refer = 10,
    Message = 11,
    Prack = 12
}

/// <summary>ResponseCode matching the Idris2 ABI tags (0-16).</summary>
public enum ResponseCode : byte
{
    Trying = 0,
    Ringing = 1,
    SessionProgress = 2,
    Ok = 3,
    MultipleChoices = 4,
    MovedPermanently = 5,
    MovedTemporarily = 6,
    BadRequest = 7,
    Unauthorized = 8,
    Forbidden = 9,
    NotFound = 10,
    MethodNotAllowed = 11,
    RequestTimeout = 12,
    BusyHere = 13,
    Decline = 14,
    ServerInternalError = 15,
    ServiceUnavailable = 16
}

/// <summary>DialogState matching the Idris2 ABI tags (0-2).</summary>
public enum DialogState : byte
{
    Early = 0,
    Confirmed = 1,
    Terminated = 2
}

File: bindings/php/src/ProvenVoip.php
<?php
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// VoIP/SIP protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Method matching the Idris2 ABI tags. */
enum Method: int
{
    case Invite = 0;
    case Ack = 1;
    case Bye = 2;
    case Cancel = 3;
    case Register = 4;
    case Options = 5;
    case Info = 6;
    case Update = 7;
    case Subscribe = 8;
    case Notify = 9;
    case Refer = 10;
    case Message = 11;
    case Prack = 12;
}

/** ResponseCode matching the Idris2 ABI tags. */
enum ResponseCode: int
{
    case Trying = 0;
    case Ringing = 1;
    case SessionProgress = 2;
    case Ok = 3;
    case MultipleChoices = 4;
    case MovedPermanently = 5;
    case MovedTemporarily = 6;
    case BadRequest = 7;
    case Unauthorized = 8;
    case Forbidden = 9;
    case NotFound = 10;
    case MethodNotAllowed = 11;
    case RequestTimeout = 12;
    case BusyHere = 13;
    case Decline = 14;
    case ServerInternalError = 15;
    case ServiceUnavailable = 16;
}

/** DialogState matching the Idris2 ABI tags. */
enum DialogState: int
{
    case Early = 0;
    case Confirmed = 1;
    case Terminated = 2;
}
